### PR TITLE
:sparkles: CRUD for Jira instances/trackers

### PR DESCRIPTION
**Pull request by gildub**
- Jira instance are now into "src/app/pages/external/jira" so future 3rd party ticketing/management systems can reside along.
- Delete notification passes the name of deleted instance using a generic Translation message. This could be easily generalized. 

image

Addresses https://github.com/konveyor/tackle2-ui/issues/588

### CODE REVIEW

File: client/src/app/FeatureFlags.ts [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export const FEATURES_ENABLED = {
-  migrationWaves: false,
+  migrationWaves: true,
```
Comment (ibolton336) [inline client/src/app/FeatureFlags.ts:2]: Looks like this wasn't intentional.

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+          />
+        )}
+      />
+
+      <Switch
+        id="insecureCommunication"
+        className="repo"
+        label="Enable insecure communication"
+        aria-label="Insecure Communication"
+        isChecked={insecureCommunication}
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:263]: Looking great, Gilles. I am wondering if this one is just left out of the form state due to lack of API support?

File: client/src/app/pages/external/jira/trackers.tsx [hunk 3]
```diff
@@ -118,17 +160,17 @@ export const JiraTrackers: React.FC = () => {
...
-                      // onClick={openCreateTrackerModal}
+                      onClick={() => setInstanceModalOpen(true)}
                     >
                       {t("actions.createNew")}
                     </Button>
                   </ToolbarItem>
                   {/* </RBAC> */}
                   {/* {jiraDropdownItems.length ? (
-                    <ToolbarItem>
+                    <ToolbarItem>setInstanceToUpdate
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/trackers.tsx:173]: Minor cleanup here.

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+            }}
+          />
+        )}
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="credentialName"
+        label={t("terms.credentials")}
+        fieldId="Credentials"
+        isRequired
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:237]: Should this be required?

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+      kind: instance?.kind,
+      credentialName: instance?.identity?.name || "",
+    },
+    resolver: yupResolver(validationSchema),
+    mode: "onChange",
+  });
+
+  const values = getValues();
+
+  const [isKeyFileRejected, setIsKeyFileRejected] = useState(false);
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:155]: Not used

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+  const [isSettingsFileRejected, setIsSettingsFileRejected] = useState(false);
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:156]: Not used

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="credentialName"
+        label={t("terms.credentials")}
+        fieldId="Credentials"
+        isRequired
+        renderInput={({ field: { value, name, onChange } }) => (
+          <SimpleSelect
+            id="credentials-select"
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:240]: Need to add: 

```
  maxHeight={DEFAULT_SELECT_MAX_HEIGHT}
  variant="typeahead"
```

This adds a default no results found when no identities are available. Also the max height helps with overflow in the case of a large number of options. Check the application-identity-form.tsx identity dropdown for an example.

**Review by ibolton336 — CHANGES_REQUESTED** [review]
Looking good. A few small changes requested.

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+        fieldId="url"
+        isRequired
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="kind"
+        label="Instance type"
+        fieldId="type-select"
+        isRequired
+        renderInput={({ field: { value, name } }) => (
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:211]:
```suggestion
        renderInput={({ field: { value, name, onChange } }) => (
```

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+            placeholderText={t("composed.selectMany", {
+              what: t("terms.instanceType").toLowerCase(),
+            })}
+            toggleAriaLabel="Type select dropdown toggle"
+            aria-label={name}
+            value={value ? toOptionLike(value, kindOptions) : undefined}
+            options={kindOptions}
+            onChange={(selection) => {
+              const selectionValue = selection as OptionWithValue<JiraKind>;
+              setValue("kind", selectionValue.value);
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:225]:
```suggestion
              onChange(selectionValue.value);
```

File: client/src/app/pages/external/jira/instance-form.tsx [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+    getValues,
+    setValue,
+    control,
+    watch,
+  } = useForm<FormValues>({
+    defaultValues: {
+      name: instance?.name || "",
+      url: instance?.url || "",
+      id: instance?.id || 0,
+      kind: instance?.kind || undefined,
```
Comment (ibolton336) [inline client/src/app/pages/external/jira/instance-form.tsx:148]:
```suggestion
      kind: instance?.kind,
```
Undefined will be assigned if kind does not exist on the instance object. That is wha optional chaining gets you.

File: client/src/app/FeatureFlags.ts [hunk 11]
```diff
@@ -1,4 +1,4 @@
...
```
[the remark this answers is not in the document]
Comment (gildub) [reply client/src/app/FeatureFlags.ts:2]: Ah yeah, just annoying to have to flip that one all the time.

[the remark this answers is not in the document]
Comment (ibolton336) [reply client/src/app/FeatureFlags.ts:2]: Yeah I know what you mean. Can probably figure out a way for git to ignore changes to it locally.

[the remark this answers is not in the document]
Comment (ibolton336) [reply client/src/app/FeatureFlags.ts]: ``` git update-index --skip-worktree client/src/app/FeatureFlags.ts ```

**Review by ibolton336 — APPROVED** [review]

**Pull request merged**